### PR TITLE
fix: require function undefined access

### PR DESCRIPTION
**Pull request by andrew-w-ross**
cache isn't always defined so `cached = cache[this.filePath]` breaks the require function. This adds a check to fix it.

### CODE REVIEW

File: src/FuseProcess.ts [hunk 1]
```diff
@@ -31,8 +31,9 @@ export class FuseProcess {
 				mdl;
 		}
 		return new Promise((resolve, reject) => {
-			let cache = (<any>require).cache, cached = cache[this.filePath], closePromise, exps;
-			if (cached) {
+			let cache = (<any>require).cache, closePromise, exps
+			const cached = cache && cache[this.filePath];
+			if (cached != null) {
```
Comment (nchanged) [inline src/FuseProcess.ts]: Null is quite a tricky thing, are u sure about it?

Comment (andrew-w-ross) [reply src/FuseProcess.ts]: Well I was just checking for null or undefined hence the `!=` and not `!==`. Unless you are referring to something else.

Comment (nchanged) [reply src/FuseProcess.ts]: What are the possible values of the `cache` variable?

Comment (andrew-w-ross) [reply src/FuseProcess.ts]: I have no idea cache is defined as any.

Comment (andrew-w-ross) [reply src/FuseProcess.ts]: @nchanged can you merge or reject and close this pr?

Comment (nchanged) [reply src/FuseProcess.ts]: Please don’t rush me on this one, I need to test it myself I just haven’t had time yet

Comment (nchanged) [reply src/FuseProcess.ts]:
```
var cache = undefined;
cache == null // true o_O
```

what's wrong with just `if (!cache) {}` it shouldn't yield falsy values

**Comment by andrew-w-ross** [comment]
Seems to have been fixed in later releases

**Pull request closed without merging**